vm/src/builtins/enumerate.rs

```diff
@@ -60,6 +61,52 @@ impl PyIter for PyEnumerate {
     }
 }
 
+#[pyclass(module = false, name = "reversed")]
+#[derive(Debug)]
+pub struct PyReverseSequenceIterator {
+    pub position: AtomicCell<isize>,
+    pub obj: PyObjectRef,
+}
+
+impl PyValue for PyReverseSequenceIterator {
+    fn class(vm: &VirtualMachine) -> &PyTypeRef {
+        &vm.ctx.types.reverse_iter_type
+    }
+}
+
+#[pyimpl(with(PyIter))]
+impl PyReverseSequenceIterator {
+    pub fn new(obj: PyObjectRef, len: isize) -> Self {
+        Self {
+            position: AtomicCell::new(len - 1),
+            obj,
+        }
+    }
+
+    #[pymethod(magic)]
+    fn length_hint(&self) -> PyResult<isize> {
+        Ok(self.position.load() + 1)
+    }
+}
+
+impl PyIter for PyReverseSequenceIterator {
+    fn next(zelf: &PyRef<Self>, vm: &VirtualMachine) -> PyResult {
+        let pos = zelf.position.fetch_sub(1);
```

Comment:
what happens if this is called forever until negatively overflow?

Reply:
I'm fixing all these methods in another pr. Currently I've just split these appart for now to make the changes a little smaller in size.

Reply:
Will you make another huge PR with the whole diff for a reference? I will also check the final result.

Reply:
I could push the changes on this branch if it would make it easier for you.

Reply:
I think seperated small PR still have benefits. So another PR would be more convenient. It will be reusable after merging this one.

For this PR, I thought I can compare other parts also, but after checking your answers about it, it doesn't look that's nessessary for now.